notebooks/maze_tuto.ipynb

```diff
@@ -0,0 +1,642 @@
+    "- to create its own rollout function to play a trained solver on a domain\n",
+    "\n",
+    "\n",
+    "Notes:\n",
+    "- In order to focus on scikit-decide use, we put some code not directly related to the library in a [separate module](./maze_utils.py) (like maze generation and display).\n",
+    "- A similar maze domain is already defined in [scikit-decide hub](https://github.com/airbus/scikit-decide/blob/master/skdecide/hub/domain/maze/maze.py) but we do not use it for the sake of this tutorial.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import io\n",
+    "from enum import Enum\n",
+    "from typing import NamedTuple, Optional, Any, List\n",
+    "from time import sleep\n",
+    "from collections import deque\n",
+    "import random\n",
+    "from math import sqrt, ceil\n",
+    "\n",
+    "from IPython.display import Markdown, display, clear_output\n",
+    "import ipywidgets as widgets\n",
+    "import matplotlib.pyplot as plt\n",
+    "from stable_baselines3 import PPO\n",
+    "\n",
+    "from skdecide import DeterministicPlanningDomain, Space, Value\n",
+    "from skdecide.builders.domain import UnrestrictedActions, Renderable\n",
+    "from skdecide.utils import rollout, match_solvers, load_registered_solver\n",
+    "from skdecide.hub.space.gym import ListSpace, EnumSpace, MultiDiscreteSpace\n",
+    "from skdecide.hub.solver.lazy_astar import LazyAstar\n",
+    "from skdecide.hub.solver.stable_baselines import StableBaseline\n",
+    "\n",
+    "# import Maze class from utility file for maze generation and display\n",
+    "from maze_utils import Maze\n",
+    "\n",
+    "# choose standard matplolib inline backend to render plots\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-09-28T13:19:59.841418Z",
+     "start_time": "2021-09-28T13:19:59.836876Z"
+    }
+   },
+   "source": [
+    "## About maze problem\n",
+    "The maze problem is about to make an agent finding the goal in a maze by going up, down, left, or right without going through walls. \n",
+    "\n",
+    "We show you such a maze by using the Maze class defined in the [maze module](./maze_utils.py). Here the agent starts at the top-left corner and the goal is at the bottom_right corner of the maze. The following colour convention is used:\n",
+    "- dark purple: walls\n",
+    "- yellow: empty cells\n",
+    "- light green: goal\n",
+    "- blue: current position"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# size of maze\n",
+    "width = 25\n",
+    "height = 19\n",
+    "# generate the maze\n",
+    "maze = Maze.generate_random_maze(width=width, height=height)\n",
+    "# starting position\n",
+    "i_start, j_start = 1, 1\n",
```

Comment:
Might make sense to store start and final positions as tuples. This would be consistent with State definition

Reply:
Yes but, maze_utils.py is not aware about State definition. Actually this notebook is somehow about wrapping an existing class (Maze) that do no know about scikit-decide, in a scikit-decide domain. that's why Maze.render cannot use State as input. I can still use tuples though, i just need to be cautious to avoid confusion between the tuple and State class...